### ISSUE TITLE
 /^.+<\?/i.test('<?') gives true in Edge 40.15063.0.0 but false in other browsers/engines

### ISSUE THREAD

**Issue by martin-honnen** [Microsoft/ChakraCore#3366]
I think in https://stackoverflow.com/questions/45122823/is-there-any-reason-why-edge-returns-true-for-xml-i-testxml-version I have found a bug with the regular expression `test` method in the Javascript engine of Edge 40.15063.0.0.

```
var xml = '<?';
var regex = /^.+<\?/i;
var test = regex.test(xml);
```
As there is no character at all before the `<` less-than sign in the input string the match `^.+` should not match anything and therefore that `test` call should return `false` and not `true` as it does in the cited Edge version.

**Comment by MizardX** [comment]
`/^.+%x/.test('%x')` displays the same bug. Tested in Edge 41.16232.1000.0

**Comment by martin-honnen** [comment]
Based on the tests added in https://github.com/Microsoft/ChakraCore/commit/a8e157afdd54177471d66e06dbfcc0ee1c36d437#diff-18608f601e3f7519b31ec8d2d2e28e25 it looks as if that fix already tries to address the posted problem with `/^.+`.

**Comment by MikeHolman** [comment]
You're right. I verified in our console host and your bug is fixed by #3201, which will ship as part of the next version of Edge.

**Comment by curtisman** [comment]
Closing the issue as this is already fixed in ChakraCore v1.6 and master
